### PR TITLE
feature/auth/change password

### PR DESCRIPTION
**Pull request by turchin-aa**

### CODE REVIEW

File: controllers/auth.controller.ts [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+}
+
+export const changePassword = async (req: Request, res: Response) => {
+  try {
+    const { oldPassword, newPassword } = req.body
+    if (!oldPassword || !newPassword) {
+      return res.status(400).json({ message: 'Invalid data' })
+    }
+    const { refreshToken } = req.cookies
+    const tokenFromDB = await findToken(refreshToken)
```
Comment (VolodymyrDevico) [inline controllers/auth.controller.ts:121]: Better use access token for this purposes

File: controllers/auth.controller.ts [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+    const { id } = candidate.toJSON()
+    const token: any = await Token.findOne({ where: { id } })
+    const { resetPasswordToken } = token.toJSON()
+
+    if (resetPasswordToken) {
+      await Token.destroy({ where: { refreshToken: token } })
+    }
+
+    const resetToken: any = generateResetToken({ id, email })
```
Comment (VolodymyrDevico) [inline controllers/auth.controller.ts:157]: don't use any type

File: db/database.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
+import { Sequelize } from 'sequelize'
+import dotenv from 'dotenv'
+
+dotenv.config()
+
+const DB: any = process.env.DB
```
Comment (VolodymyrDevico) [inline db/database.ts:6]: const { ... } = process.env

File: dist/index.js [hunk 4]
```diff
@@ -4,13 +4,17 @@ var __importDefault = (this && this.__importDefault) || function (mod) {
 };
```
Comment (VolodymyrDevico) [inline dist/index.js:4]: remove dist

**Review by VolodymyrDevico — CHANGES_REQUESTED** [review]

**Review by ntnshpig — APPROVED** [review]

**Pull request merged**